### PR TITLE
Update CICE (2024-11) including pio_finalize bug fix

### LINKED ISSUE #94: Coupled weather model forecasts fail after large # of file writes when CICE is compiled using PIO
## Description
Using CICE in a S2S configuration in ufs-weather-model causes failures after a large number of CICE file (restart and/or history) writes (500-700ish) when CICE is compiled with PIO but not with NetCDF. The failure always happens on a CICE process. The current work around for weather model regression tests have been to set `export I_MPI_SHM_HEAP_VSIZE=16384` in the job submission script, but this is not a long-term solution.

### To Reproduce:
1. Compile weather model with ATM+ICE+OCN on Hera, Gaea, or WCOSS2. Have used multiple different weather model regression test configurations and resolutions (cpld_control_c48, cpld_control_nowave_noaero_p8) and stack-stack versions/intel compilers (2021v2023) with similar results. 
2. Either run very long simulations with infrequent output or shorter simulations with high frequency output.
3. Experience failure after 500-700 files written.

## Additional context
Cause of issue first reported in [weather model issue 2320](https://github.com/ufs-community/ufs-weather-model/issues/2320) 

I've also tried all possible options of restart/history_format in ice_in and the failure is always the same. 

## Output
 On Hera the failure looks like:

```
73: Assertion failed in file ../../src/mpid/ch4/src/intel/ch4_shm_coll.c at line 2266: comm->shm_numa_layout[my_numa_node].base_addr
73: /apps/oneapi/mpi/2021.5.1/lib/release/libmpi.so.12(MPL_backtrace_show+0x1c) [0x150a7a430bcc]
73: /apps/oneapi/mpi/2021.5.1/lib/release/libmpi.so.12(MPIR_Assert_fail+0x21) [0x150a79e0adf1]
73: /apps/oneapi/mpi/2021.5.1/lib/release/libmpi.so.12(+0x2b1eb9) [0x150a79ad9eb9]
73: /apps/oneapi/mpi/2021.5.1/lib/release/libmpi.so.12(+0x176584) [0x150a7999e584]
73: /apps/oneapi/mpi/2021.5.1/lib/release/libmpi.so.12(+0x17a9f9) [0x150a799a29f9]
73: /apps/oneapi/mpi/2021.5.1/lib/release/libmpi.so.12(+0x199a60) [0x150a799c1a60]
73: /apps/oneapi/mpi/2021.5.1/lib/release/libmpi.so.12(+0x1717ec) [0x150a799997ec]
73: /apps/oneapi/mpi/2021.5.1/lib/release/libmpi.so.12(+0x2b4387) [0x150a79adc387]
73: /apps/oneapi/mpi/2021.5.1/lib/release/libmpi.so.12(PMPI_Allreduce+0x561) [0x150a799376e1]
73: /apps/oneapi/mpi/2021.5.1/lib/release/libmpi.so.12(MPI_File_open+0x17d) [0x150a7a4492bd]
73: /scratch1/NCEPDEV/nems/role.epic/spack-stack/spack-stack-1.6.0/envs/unified-env-rocky8/install/intel/2021.5.0/parallel-netcdf-1.12.2-cwokdeb/lib/libpnetcdf.so.4(ncmpio_create+0x199) [0x150a73e592c9]
73: /scratch1/NCEPDEV/nems/role.epic/spack-stack/spack-stack-1.6.0/envs/unified-env-rocky8/install/intel/2021.5.0/parallel-netcdf-1.12.2-cwokdeb/lib/libpnetcdf.so.4(ncmpi_create+0x4e7) [0x150a73daf4a7]
73: /scratch1/NCEPDEV/nems/role.epic/spack-stack/spack-stack-1.6.0/envs/unified-env-rocky8/install/intel/2021.5.0/parallelio-2.5.10-2wulfav/lib/libpioc.so(PIOc_createfile_int+0x2e6) [0x150a7c436696]
73: /scratch1/NCEPDEV/nems/role.epic/spack-stack/spack-stack-1.6.0/envs/unified-env-rocky8/install/intel/2021.5.0/parallelio-2.5.10-2wulfav/lib/libpioc.so(PIOc_createfile+0x41) [0x150a7c432451]
73: /scratch1/NCEPDEV/nems/role.epic/spack-stack/spack-stack-1.6.0/envs/unified-env-rocky8/install/intel/2021.5.0/parallelio-2.5.10-2wulfav/lib/libpiof.so(piolib_mod_mp_createfile_+0x25e) [0x150a7c1caabe]
```
On WCOSS2 and Gaea the error looks like

```
17: MPICH ERROR [Rank 17] [job id 135188771.0] [Mon Oct  7 17:37:30 2024] [c5n1294] - Abort(806965007) (rank 17 in comm 0): Fatal error in PMPI_Comm_split: Other MPI error, error stack:
17: PMPI_Comm_split(513)................: MPI_Comm_split(comm=0xc400314e, color=1, key=0, new_comm=0x7ffe5bbb2d74) failed
17: PMPI_Comm_split(494)................:
17: MPIR_Comm_split_impl(268)...........:
17: MPIR_Get_contextid_sparse_group(610): Too many communicators (0/2048 free on this process; ignore_id=0)
```

### PR DESCRIPTION
**Pull request by NickSzapiro-NOAA**
## Commit Queue Requirements:

- [x] Fill out all sections of this template.
- [x] All sub component pull requests have been reviewed by their code managers.
- [x] Run the full Intel+GNU RT suite (compared to current baselines) on either Hera/Derecho/Hercules
- [x] Commit 'test_changes.list' from previous step
---
## Description:

Merge authoritative CICE-Consortium:main into EMC fork.

This also enables existing hist_suffix to CICE tests/parm/ice_in.IN that can be used to create unique filenames when different instantaneous history output frequencies write for the same output time. This has no impact on regression tests as CICE hist_avg=.true.

### Commit Message:

```
* UFSWM - Update CICE to Consortium/main (2024-11). Expose hist_suffix for CICE ice_in
  * CICE - Update CICE to Consortium/main (2024-11)
```

### Priority:

* High: Bugfix. Resolves model crash when trying to output many CICE history files
* High: Needed by https://github.com/NOAA-EMC/global-workflow/pull/3032

### UFSWM:

* Closes https://github.com/NOAA-EMC/CICE/issues/94 (and presumably https://github.com/ufs-community/ufs-weather-model/issues/2320)
* Closes #https://github.com/ufs-community/ufs-weather-model/issues/2509

### Sub component Pull Requests:

* CICE: https://github.com/NOAA-EMC/CICE/pull/96

### UFSWM Blocking Dependencies:

* None

---
### Regression Test Changes (Please commit test_changes.list):

* PR Updates/Changes Baselines.
Answer changing for all tests using CICE, from change in lateral melt in https://github.com/CICE-Consortium/Icepack/pull/495

### Input data Changes:

* None.

### Library Changes/Upgrades:

* No Updates
  
---

## Testing Log:
- RDHPCS
  - [x] Hera
  - [x] Orion
  - [x] Hercules
  - [x] Jet
  - [x] Gaea
  - [x] Derecho
- WCOSS2
  - [x] Dogwood/Cactus
  - [x] Acorn
- [ ] CI
- [ ] opnReqTest (complete task if unnecessary)

### CODE REVIEW

File: tests/logs/RegressionTests_hercules.log [hunk 1]
```diff
@@ -31,400 +31,397 @@ Submodule hashes used in testing:
...
+
+PASS -- COMPILE atm_dyn32_intel [09:39, 09:38](1 warnings,1 remarks)
+PASS -- TEST control_flake_intel [03:02, 02:51] (712540 MB)
+PASS -- TEST control_CubedSphereGrid_intel [03:04, 02:41] (1599404 MB)
+PASS -- TEST control_CubedSphereGrid_parallel_intel [04:08, 03:40] (1592960 MB)
+PASS -- TEST control_latlon_intel [03:05, 02:45] (1582116 MB)
+PASS -- TEST control_wrtGauss_netcdf_parallel_intel [03:12, 02:44] (1592800 MB)
+PASS -- TEST control_c48_intel [10:53, 10:34] (1704104 MB)
+PASS -- TEST control_c48.v2.sfc_intel [06:13, 06:01] (833868 MB)
+PASS -- TEST control_c192_intel [10:41, 10:11] (1754152 MB)
```
Comment (BrianCurtis-NOAA) [inline tests/logs/RegressionTests_hercules.log:115]: What happened to the lnd_iau_intel test?

Comment (NickSzapiro-NOAA) [reply tests/logs/RegressionTests_hercules.log:115]: I don't know...@jkbk2004 ?

Comment (jkbk2004) [reply tests/logs/RegressionTests_hercules.log:115]: pushed hercules log again: test script sometimes not going thru fully between rocoto and log created on hercules.

**Comment by jkbk2004** [comment]
It's good to start merging process.

**Comment by NickSzapiro-NOAA** [comment]
@jkbk2004 @BrianCurtis-NOAA I made merge commit into EMC/CICE. Should be ready

**Review by jkbk2004 — APPROVED** [review]

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Pull request merged**